helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/DefaultZooKeeperClient.java

```diff
@@ -96,7 +96,7 @@ public void process(WatchedEvent event) {
       @Override
       public void stateChanged(CuratorFramework client, ConnectionState newState) {
         log.info("DefaultZooKeeperClient connection state change - {}", newState);
-        if (newState == ConnectionState.RECONNECTED) {
+        if (newState == ConnectionState.RECONNECTED || newState == ConnectionState.CONNECTED) {
```

Comment:
I think that the only thing that `checkClusterIdExists` and related methods cares about is that the connection is active, so while both of these values of the enum should definitely be considered as a condition for calling `checkClusterIdExists`, it looks like this check could be simplified as `if (newState.isConnected()) ...`